SimTracker/SiPixelDigitizer/plugins/SiPixelChargeReweightingAlgorithm.cc

```diff
@@ -533,32 +536,224 @@ int SiPixelChargeReweightingAlgorithm::PixelTempRewgt2D(int id_in, int id_rewgt,
+      theDigiSignal[chan] += SiPixelDigitizerAlgorithm::Amplitude(corresponding_charge, corresponding_charge);
+      chargeBefore += corresponding_charge;
+      if (loopDigi->row() < irow_min)
+        irow_min = loopDigi->row();
+      if (loopDigi->row() > irow_max)
+        irow_max = loopDigi->row();
+      if (loopDigi->column() < icol_min)
+        icol_min = loopDigi->column();
+      if (loopDigi->column() > icol_max)
+        icol_max = loopDigi->column();
     }
-    std::cout << std::endl;
   }
-  std::cout.copyfmt(std::ios(nullptr));
+  // end loop on digis
+
+  LocalVector direction = loopTempSH.exitPoint() - loopTempSH.entryPoint();
+  LocalPoint hitEntryPoint = loopTempSH.entryPoint();
+  float trajectoryScaleToPosition = std::abs(hitEntryPoint.z() / direction.z());
+  LocalPoint hitPosition = hitEntryPoint + trajectoryScaleToPosition * direction;
+
+  // addition as now the hit himself is not available
+  // https://github.com/cms-sw/cmssw/blob/master/SimDataFormats/TrackingHit/interface/PSimHit.h#L52
+  LocalPoint hitLocalPosition = hitEntryPoint + 0.5 * direction;
+  MeasurementPoint hitPositionPixel = topol->measurementPosition(hitLocalPosition);
+  std::pair<int, int> hitPixel =
+      std::pair<int, int>(int(floor(hitPositionPixel.x())), int(floor(hitPositionPixel.y())));
+
+  MeasurementPoint originPixel = MeasurementPoint(hitPixel.first - THX + 0.5, hitPixel.second - THY + 0.5);
+  LocalPoint origin = topol->localPosition(originPixel);
+
+  MeasurementPoint hitEntryPointPixel = topol->measurementPosition(loopTempSH.entryPoint());
+  MeasurementPoint hitExitPointPixel = topol->measurementPosition(loopTempSH.exitPoint());
+  std::pair<int, int> entryPixel =
+      std::pair<int, int>(int(floor(hitEntryPointPixel.x())), int(floor(hitEntryPointPixel.y())));
+  std::pair<int, int> exitPixel =
+      std::pair<int, int>(int(floor(hitExitPointPixel.x())), int(floor(hitExitPointPixel.y())));
+
+  int hitcol_min, hitcol_max, hitrow_min, hitrow_max;
+  if (entryPixel.first > exitPixel.first) {
+    hitrow_min = exitPixel.first;
+    hitrow_max = entryPixel.first;
+  } else {
+    hitrow_min = entryPixel.first;
+    hitrow_max = exitPixel.first;
+  }
+
+  if (entryPixel.second > exitPixel.second) {
+    hitcol_min = exitPixel.second;
+    hitcol_max = entryPixel.second;
+  } else {
+    hitcol_min = entryPixel.second;
+    hitcol_max = exitPixel.second;
+  }
+
+  if (!(irow_min <= hitrow_max && irow_max >= hitrow_min && icol_min <= hitcol_max && icol_max >= hitcol_min)) {
+    // The clusters do not have an overlap, hence the hit is NOT reweighted
+    return false;
+  }
+
+  track[0] = (hitPosition.x() - origin.x()) * cmToMicrons;
+  track[1] = (hitPosition.y() - origin.y()) * cmToMicrons;
+  track[2] = 0.0f;  //Middle of sensor is origin for Z-axis
+  track[3] = direction.x();
+  track[4] = direction.y();
+  track[5] = direction.z();
+
+  array_2d pixrewgt(boost::extents[TXSIZE][TYSIZE]);
+
+  /*
```

Comment:
I don't want to retrigger another set of corrections and signatures, but this commented out code should be removed, even in a follow up PR